src/modm/math/geometry/vector.hpp

```diff
@@ -11,141 +12,552 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 // ----------------------------------------------------------------------------
+#pragma once
 
-#ifndef MODM_VECTOR_HPP
-#define MODM_VECTOR_HPP
-
+#include <array>
 #include <cmath>
-#include <stdint.h>
+#include <concepts>
+#include <functional>
+#include <numeric>
+#include <ranges>
 
-#include <modm/math/matrix.hpp>
 #include <modm/math/utils/arithmetic_traits.hpp>
+#include <modm/math/utils/integer_traits.hpp>
+#include <modm/io/iostream.hpp>
+
+#include "../matrix.hpp"
+#include "angle.hpp"
 
 namespace modm
 {
-	// forward declaration
-	template<typename T, uint8_t W, uint8_t H> class Matrix;
+
+/**
+ * \brief	Class for handling common vector operations
+ *
+ * Arithmetic Operations:
+ * \code
+ *	+ : plus translation
+*	- : minus translation
+*	* : scalar multiplication / dot product
+*	/ : scalar division
+*	^ : cross product / determinant
+*	~ : perpendicular / orthogonal
+* \endcode
+*
+* \author	Fabian Greif
+* \author	Niklas Hauser
+* \author	Thomas Sommer
+* \ingroup	modm_math_geometry
+*/
+template<class T, std::size_t N>
+requires (N > 0)
+class Vector : public std::array<T, N>
+{
```

Comment:
The inheritance on std::array adds lot's of comfort to modm::Vector, f.e. passing them directly to the std::ranges API.